src/components/Menu/Menu.tsx

```diff
@@ -98,13 +150,21 @@ const MenuPlainButton = ({ className, ...other }: MenuPlainButtonProps) => {
  * @param props Props used on the set of menu items
  * @see https://headlessui.com/react/menu#menu-items
  */
-const MenuItems = (props: MenuItemsProps) => (
-  <HeadlessMenu.Items
-    as={PopoverContainer}
-    className={clsx(styles['menu__popover'])}
-    {...props}
-  />
-);
+const MenuItems = (props: MenuItemsProps) => {
+  const { setPopperElement, popperStyles, popperAttributes } =
+    useContext(MenuContext);
+  const optionProps = {
+    as: PopoverContainer,
+    ref: setPopperElement,
+    style: popperStyles,
+    ...props,
+    ...popperAttributes,
+  };
+
+  return (
+    <>{createPortal(<HeadlessMenu.Items {...optionProps} />, document.body)}</>
```

Comment:
Does this mean that the headlesss menu items will always be in the document body?

Reply:
@jinlee93 Correct. by using portal on `<body>` we can make sure the z-index is always correct for any popover. There are a few side-effects: some of the snapshots will change b/c the dom content won't be adjacent to the trigger anymore, and the size of some chromatic tests changed b/c they were based on relative positioning of content. since the popover is not directly next to the trigger, the visual diffs had to change a bit to show the menu.

Reply:
Could this potentially cause any tabbing / keyboard accessibility issues?

Reply:
> Could this potentially cause any tabbing / keyboard accessibility issues?

Valid concern. HeadlessUI manages the focus trap from the trigger to the popover, accounting for cases like this. So, whether the popover is adjacent to the trigger in the DOM, or portal'd somewhere else, it will work and return to the trigger if <kbd>ESC</kbd> is hit, for example

Reply:
Not sure if this is relevant, but what about having a persistent `<div>` as a child of the document body to place the popper elements? ref: https://popper.js.org/docs/v2/performance/#attaching-elements-to-the-dom

Reply:
I thought about adding a container somewhere that this will target, but it's tricky to make it generic. We wouldn't want people to have to add some particular container whenever they use a popover-like component. What if they forget to? or change their mind?

`document.body` is guaranteed to exist and using portal here should prevent any issues with react itself (this will put it outside the app container on that page)

One way to solve for this is if we introduce some kind of "EDS app" component which we'd encourage people to use around their whole app. We could then enforce such a container. Otherwise we could put the target next to the trigger, but that would reintroduce the overflow issues ...

I'll keep that in mind when I see how some consuming apps work with a published alpha release

Reply:
Makes sense to me, thanks for the explanation